src/app/models/prediction.py

```diff
@@ -38,14 +47,22 @@ async def get_prediction(self, user_name) -> list[dict]:
 
         """
         async with self.session:
-            query: Select = select(dbPrediction)
-            query = query.select_from(dbPrediction)
-            query = query.where(dbPrediction.name == user_name)
+            prediction_db: dbPrediction = aliased(dbPrediction, name="prediction_db")
+            query: Select = select(prediction_db)
+            query = query.select_from(prediction_db)
+            query = query.where(
+                text("prediction_db.name = :name").bindparams(name=user_name)
+            )  # prevent sql injection
             result: Result = await self.session.execute(query)
             await self.session.commit()
 
-        # transform output to json
         output = result.mappings().all()
-        output = [o.get("Prediction") for o in output]
-        output = jsonable_encoder(output)
+        # verify prediction data exists
+        if all("Prediction" not in o for o in output):
+            output = [object_as_dict(o["prediction_db"]) for o in output]
+            output = jsonable_encoder(output)
+        # verify output is not empty list of none
+        if all(o is None for o in output):
+            return None
+
         return output

```

Comment:
what are the reasons for these changes

Reply:
if the returned mappings is None we need logic to return None before getting to https://github.com/Bot-detector/public-api/blob/4c1180aa78f47297df01280f7c21c719b9b79d35/src/app/models/prediction.py#L62

otherwise you will get a list with None [None] so you need this to remediate.  

looking over we should just check the output value is not None otherwise return none so it never gets to object processing [None] list condition.

Reply:
resolved with this commit - 
[prediction user_name to name, remove bind, revert list match, fix tes…](https://github.com/Bot-detector/public-api/pull/25/commits/d6ea27becdde54111f6d60eab2770e3c49e73839)